.github/workflows/ci.yml

```diff
@@ -0,0 +1,34 @@
+name: CI
+on:
+  push:
+    branches: [main, 'feat/**']
+    paths-ignore:
+      - '**.md' # Do not need to run CI for markdown changes.
+  pull_request:
+    branches: [main, 'feat/**']
+    paths-ignore:
+      - '**.md'
+
+jobs:
+  build-and-test:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/ci
+        with:
+          run_tests: true
+          run_contract_tests: ${{ matrix.os != 'windows-latest' }}
```

Comment:
Consider adding a comment for this special case

Reply:
As this is a merge forward I can address that in a micro-pr after merging this one.